Jenkinsfile

```
@@ -19,63 +25,112 @@ pipeline {
             description: 'The XTC tools version'
         )
     }
-    environment {
-        REPO = 'lib_mic_array'
-    }
     stages {
         stage('Build and Docs') {
             parallel {
-                stage('Build Docs') {
-                    agent { label "docker" }
-                    environment { XMOSDOC_VERSION = "v4.0" }
+                stage('Documentation') {
+                    agent {
+                        label "docker"
+                    }
                     steps {
-                        checkout scm
-                        sh 'git submodule update --init --recursive --depth 1'
-                        sh "docker pull ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION"
-                        sh """docker run -u "\$(id -u):\$(id -g)" \
-                            --rm \
-                            -v ${WORKSPACE}:/build \
-                            ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION -v"""
-                        archiveArtifacts artifacts: "doc/_build/**", allowEmptyArchive: true
+                        dir("${REPO}") {
+                            warnError("Docs") {
+                                checkout scm
+                                sh "docker pull ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION"
+                                sh """docker run -u "\$(id -u):\$(id -g)" \
+                                      --rm \
+                                      -v \$(pwd):/build \
+                                      ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION -v html latex"""
+
+                                // Zip and archive doc files
+                                zip dir: "doc/_build/html", zipFile: "${REPO}_docs_html.zip"
+                                archiveArtifacts artifacts: "${REPO}_docs_html.zip"
+                                archiveArtifacts artifacts: "doc/_build/pdf/${REPO}*.pdf"
+                            }
+                        }
                     }
                     post {
                         cleanup {
                             xcoreCleanSandbox()
                         }
                     }
                 }
-                stage('Basic tests') {
-                    when {
-                        expression { !env.GH_LABEL_DOC_ONLY.toBoolean() }
+                stage('XCommon build') {
+                    agent {
+                        label "x86_64 && linux"
+                    }
+                    steps {
+                        dir("${REPO}") {
+                            checkout scm
+                            dir("tests") {
+                                withTools(params.TOOLS_VERSION) {
+                                    sh 'cmake -B build -G "Unix Makefiles"'
+                                    // Note no -B build so builds the xcommon Makefile
```

```suggestion
                                    // Note no -C build so builds the xcommon Makefile
```

Tiniest of typos, but could be confusing because -B build is the cmake option.